### ISSUE TITLE
Reduce dependency on `pydantic._internal`

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#835]
See https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/834.

We currently have these imports
```python
from pydantic._internal import _generate_schema
from pydantic._internal import _typing_extra
from pydantic._internal._config import ConfigWrapper
```

I think we can at least get rid of the specific `_typing_extra.add_module_globals` function that changed when passing from 2.8.2 to 2.9.0 (https://github.com/pydantic/pydantic/commit/8dc0ce3c626d49d94ce688ffc450abf2b491aff3), and maybe we can also reduce dependency on `ConfigWrapper`. Whether we can also avoid depending on `_generate_schema.GenerateSchema` is TBD.

**Comment by tcompa** [comment]
Ref:
* https://github.com/pydantic/pydantic/issues/1391
* https://github.com/JoshuaLeivers/pydantic/pull/1

**Comment by jluethi** [comment]
See resulting dependency hell here: https://github.com/fractal-napari-plugins-collection/napari-ome-zarr-navigator/issues/29

But there will be other solutions to solve this than changing things in fractal-tasks-core
=> The conclusion is not that we need a quick fix, but that we'll need to eventually relax this to avoid similar issues

**Comment by jluethi** [comment]
Great, thanks a lot @tcompa and then let's just keep an eye on the upstream developments here before we do more work on the schema & pydantic version!

**Comment by jluethi** [comment]
Agreed! Let's have manifest creation in a separate package eventually (by fractal-tasks-core 2.0) and longer term, let's try to avoid this coming with a pydantic upper bound. A lower bound for pydantic seems necessary though.
